File: algoritmosDeOrdenamiento/Lista_personalizada.py
# Ejemplo lista personalizada con Python
class CustomList:       
    def __init__(self):     
        self.list = []

    def add(self, element):     # Método para agregar un elemento a la lista
        self.list.append(element)

    def remove(self, element):  # Método para remover un elemento de la lista
        self.list.remove(element)

    def size(self):            # Método para obtener el tamaño de la lista
        return len(self.list)

    def contains(self, element):    # Método para verificar si un elemento está en la lista
        return element in self.list

custom_list = CustomList()   # Crear una lista personalizada
custom_list.add(1)          # Agregar elementos a la lista
custom_list.add(2)
custom_list.add(3)
print("Size of list:", custom_list.size())  
print("Contains 2:", custom_list.contains(2))
custom_list.remove(2)
print("Contains 2 after removal:", custom_list.contains(2))

File: algoritmosDeOrdenamiento/Lista_personalizada.js
// Ejemplo lista personalizada Javascript
class CustomList {     
    constructor() {
        this.list = [];
    }

    add(element) {      // Agregar un elemento a la lista
        this.list.push(element);
    }

    remove(element) {   // Remover un elemento de la lista
        const index = this.list.indexOf(element);
        if (index !== -1) {
            this.list.splice(index, 1);
        }
    }

    size() {        // Obtener el tamaño de la lista
        return this.list.length;
    }

    contains(element) {     // Verificar si un elemento está en la lista
        return this.list.includes(element);
    }
}

let customList = new CustomList();
customList.add(1);
customList.add(2);
customList.add(3);
console.log("Size of list:", customList.size());
console.log("Contains 2:", customList.contains(2));
customList.remove(2);
console.log("Contains 2 after removal:", customList.contains(2));
